packages/admin/admin/src/dataGrid/excelExport/DataGridExcelExportButton.tsx

```diff
@@ -0,0 +1,44 @@
+import { Button, CircularProgress } from "@mui/material";
+import * as React from "react";
+import { FormattedMessage } from "react-intl";
+
+import { FileIcon } from "../../fileIcons/FileIcon";
+import { ExportApi } from "./useDataGridExcelExport";
+
+interface Props {
+    exportApi: ExportApi;
+    onClick?: () => void;
+    loadingComponent?: React.ReactNode;
+    onError?: () => void;
+}
+
+export const DataGridExcelExportButton: React.FunctionComponent<Props> = ({ onClick, children, exportApi, loadingComponent, onError }) => {
+    const onClickButtonPressed = () => {
+        if (onClick != null) {
+            onClick();
+        } else {
+            exportApi.exportGrid();
+        }
+    };
+
+    const { loading, error } = exportApi;
+
+    if (error && onError) {
+        onError();
+    }
+
+    return (
+        <Button
+            onClick={onClickButtonPressed}
+            startIcon={
+                loading ? <>{loadingComponent ? loadingComponent : <CircularProgress size={23} />}</> : <FileIcon fileType="application/msexcel" />
```

Comment:
We actually have our own Loading icon: https://github.com/vivid-planet/comet/blob/0f2794e722d055af690a14a91dfaa65d264f366a/packages/admin/admin/src/common/Loading.tsx

I'd prefer it if you used this one.